src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -225,32 +232,42 @@ internal void OnBeforeWrite(TokenCacheNotificationArgs args)
                 instanceDiscoveryMetadataEntry);
 
             var msalAccessTokenCacheItem =
-                new MsalAccessTokenCacheItem(preferredEnvironmentHost, requestParams.ClientId, response, tenantId)
+                new MsalAccessTokenCacheItem(preferredEnvironmentHost, requestParams.ClientId, response, tenantId, subject)
                 {
-                    UserAssertionHash = requestParams.UserAssertion?.AssertionHash
+                    UserAssertionHash = requestParams.UserAssertion?.AssertionHash,
+                    IsAdfs = requestParams.AuthorityInfo.AuthorityType == AppConfig.AuthorityType.Adfs
```

Comment:
>requestParams [](start = 29, length = 13)

consider extracting this as an independent bool given this is being used in other places. e.g. the line below where setting IsAdfs to what ever the AT token value has set it to makes the code less readable imo.